### DISCUSSION TITLE
User data flow documentation

### DISCUSSION THREAD

**Discussion by bnomis** [aquasecurity/trivy#4606] [Documentation]
### Description

I couldn't find documentation on how or if any user data (source code etc.) flows over the network when using Trivy. For example, does Trivy send any user data to a server for analysis or is it all local?

### Suggestions

I suggest adding a page to the documentation to make the data flow clear and explicit.

**Comment by knqyf263** [comment]
>For example, does Trivy send any user data to a server for analysis or is it all local?

It Is processed locally in standalone mode.

**Reply by bnomis** [reply]
Thanks for the reply. 

I found the mode documentation:

- https://aquasecurity.github.io/trivy/v0.42/docs/references/modes/standalone/
- https://aquasecurity.github.io/trivy/v0.42/docs/references/modes/client-server/ 

It looks like by default no user data leaves the user's environment. The user would have to explicity use a server outside of their environment for data to leave their environment. 

There's no data flowing back to Aqua etc. behind the scenes.

Is this correct?

**Reply by itaysk** [reply]
yes

**Reply by AnaisUrlichs** [reply]
@bnomis do you think this is clear enough in the documentation or should we add more information/make it more explicit?
Your input is greatly appreciated

**Reply by bnomis** [reply]
Thanks for the reply.
It was not clear for me in the documentation. I searched for "data flow" or "privacy" etc.
I suggest making a page/section to explicity state how Aqua handles user data, what is sent where etc. Keeping everything inside a client's environment is a big plus which I think should be highlighted. 
This page from Snyk is a good example https://docs.snyk.io/more-info/how-snyk-handles-your-data

**Reply by itaysk** [reply]
Just to clarify: Snyk is a commercial SaaS product, Trivy is an open source scanning tool. So the motivation for a doc page is entirely different. We could add an FAQ item but honestly I'm not sure many will encounter it.

Also, I noticed you call this tool Aqua which is a different thing: Aqua is also a commercial SaaS product that is powered by Trivy. Some differences are covered [here](https://github.com/aquasecurity/resources/blob/main/trivy-aqua.md)

**Reply by bnomis** [reply]
I was using Aqua to mean "the organisation behind Trivy"

**Reply by AnaisUrlichs** [reply]
Hi @bnomis thank you for clarifying and your input, I will create a PR in the next days to provide more details/make it easier to find the information in the docs

**Reply by alatalo** [reply]
I was looking for the same information from the docs and could not find the mentioned links (under Docs > References > Modes >). I suggest a topic in FAQ with a clear Privacy title. Now that I would've found in a few seconds :)

**Reply by AnaisUrlichs** [reply]
Hi @alatalo the Aqua Enterprise produce is disconnected from the open source tool Trivy. This is similar to the relationship many companies have with ArgoCD, they use and build on ArgoCD but they are not a default extension of Argo.

While many of the core maintainers of Trivy are employed or affiliated with Aqua, it is not connected to the commercial product. Aqua does not know who is using Trivy where and why, different to other free or semi-open source projects. We have a disclaimer in README https://github.com/aquasecurity/trivy
Please let us know if you are missing information.
